### ISSUE TITLE
Add offset argument to limit

### ISSUE THREAD

**Issue by pauldix** [influxdata/platform#52]
We want to be able to paginate through result sets. For example if we're pulling back distinct tag values, we'd like to be able to do this:

```
from(db:"db")
    |> range(start:-5000h)
    |> filter(fn:(r) => r._measurement == "cpu")
    |> group(by:["host"])
    |> distinct(column:"host")
    |> group(none:true)
    |> limit(n:100, offset:300)
```

@121watts will need this to really complete the schema explorer

**Comment by nathanielc** [comment]
The intended behavior is to page through results. I want to be clear about how offset will work.

Limit is a simple operation that does not modify a table's partition key, so it produces exactly one output table for each input table. This means that offset would apply to the number of records in each table, like `n` does. So offset simply skips the first `300` rows and then outputs the rest.

In this use case it works well since the `group(none:true)` step ensure there is only a single output table.

We may be able to optimize this later, but the initial implementation will simply discard the offset results.

**Comment by pauldix** [comment]
Yeah, we also need to think about how to limit the number of tables and be able to page through those. Different function name? Or maybe just another argument to `limit` like `limit(n:100, tables:true)`?

**Comment by nathanielc** [comment]
Agreed.

For both tables and records we need to ensure the sort order is stable. Sorting rows is already possible with the `sort` function. I have opened #58 to discuss table sort order.
